### PR TITLE
feat:  Add unflatten method

### PR DESCRIPTION
**Pull request by GordonSmith**
fixes https://github.com/hpcc-systems/hpcc-js-wasm/issues/149

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@borango - can you review?

File: src-cpp/graphviz/main.cpp [hunk 1]
```diff
@@ -136,5 +137,134 @@ const char *Graphviz::layout(const char *src, const char *format, const char *en
...
+    snprintf(buf, sizeof(buf), "%d", newlen);
+    agxset(e, sym, buf);
+}
+
+static Agsym_t *bindedgeattr(Agraph_t *g, const char *str)
+{
+    return agattr(g, AGEDGE, const_cast<char *>(str), "");
+}
+
+#include <iostream>
```
Comment (GordonSmith) [inline src-cpp/graphviz/main.cpp:182]: Remove

File: src-ts/graphviz.ts [hunk 2]
```diff
@@ -148,6 +148,38 @@ export class Graphviz {
...
+        let errorMsg = "";
+        try {
+            try {
+                retVal = graphViz.unflatten(dotSource, l, f, c);
+            } catch (e: any) {
+                errorMsg = e.message;
+            };
+            errorMsg = graphViz.lastError() || errorMsg;
+        } finally {
+            this._module.destroy(graphViz);
```
Comment (GordonSmith) [inline src-ts/graphviz.ts:181]: Add unload on error:

**Review by borango — APPROVED** [review]
Sure. You are amazing, @GordonSmith .

Have not set up a build environment for hpcc-wasm, so my review is theoretical.

I recognize and confirm the idl change and the unit test.

Can't say much about the c sources changes. Hope this helps!

**Pull request merged**